### PR TITLE
[FLINK-7813] [metrics] Replace MetricRegistryThreadFactory

### PR DESCRIPTION
**Pull request by zentol**
## What is the purpose of the change

This PR modifies the MetricRegistry to use the ExecutorThreadFactory instead of its own MetricRegistrThreadFactory. The implementations are virtually identical, so this primarily serves
code-deduplication.

### CODE REVIEW

**Comment by GJL** [comment]
👍 

I reviewed the code and can confirm that the implementations are virtually identical.

File: flink-runtime/src/main/java/org/apache/flink/runtime/metrics/MetricRegistry.java [hunk 1]
```diff
@@ -402,25 +401,4 @@ public void run() {
...
-		private final ThreadGroup group;
-		private final AtomicInteger threadNumber = new AtomicInteger(1);
-
-		MetricRegistryThreadFactory() {
-			SecurityManager s = System.getSecurityManager();
-			group = (s != null) ? s.getThreadGroup() : Thread.currentThread().getThreadGroup();
-		}
-
-		public Thread newThread(Runnable r) {
-			Thread t = new Thread(group, r, "Flink-MetricRegistry-" + threadNumber.getAndIncrement(), 0);
```
Comment (GJL) [inline flink-runtime/src/main/java/org/apache/flink/runtime/metrics/MetricRegistry.java]: The thread name will be different after this change. New thread naming scheme is `<poolName>-thread-<threadNumber>`.

Comment (zentol) [reply flink-runtime/src/main/java/org/apache/flink/runtime/metrics/MetricRegistry.java]: That's true. It's not important imo; in fact I would argue that it's actually an argument for merging it to make the names more consistent.

**Comment by zentol** [comment]
merging.

**Pull request closed without merging**